x-pack/solutions/security/plugins/security_solution/public/detections/components/alerts_table/grouping_settings/default_group_panel_renderers.tsx

```diff
@@ -24,7 +24,17 @@ import type { GenericBuckets } from '../../../../../common/search_strategy';
 import { PopoverItems } from '../../../../common/components/popover_items';
 import { COLUMN_TAGS } from '../../../../detection_engine/common/translations';
 
-export const renderGroupPanel: GroupPanelRenderer<AlertsGroupingAggregation> = (
+/**
+ * Returns renderers to be used in the `buttonContent` property of the EuiAccordion component used within the kbn-grouping package.
+ * It handles custom renders for the following fields:
+ * - kibana.alert.rule.name
+ * - host.name
+ * - user.name
+ * - source.ip
+ * For all the other fields the default renderer managed within the kbn-grouping package will be used.
+ * These go hand in hand with defaultGroupingOptions and defaultGroupStats.
+ */
+export const defaultGroupPanelRenderers: GroupPanelRenderer<AlertsGroupingAggregation> = (
```

Comment:
why is this, a function that returns jsx based on some values that are props in another component, not just a component itself? ha

Reply:
Because that's how the grouping table was working, I didn't want to change toooo much 😆 
I just moved the code to be able to pass via prop. Just trying to limit the risk of breaking things 🤣